src/Analyzer/Resolve/resolveFunction.cpp

```diff
@@ -247,49 +249,162 @@ ProjectionNames QueryAnalyzer::resolveFunction(QueryTreeNodePtr & node, Identifi
+                        scope,
+                        false /*allow_lambda_expression*/,
+                        false /*allow_table_expression*/)[0];
+
+                    result_projection_names[0].insert(strlen("multiIf") + 1, arguments_projection_names[0] + ", " + second_argument + ", ");
+
+                    if (if_function_arguments[1]->getNodeType() != QueryTreeNodeType::IDENTIFIER &&
+                        !function_query_node->getResultType()->equals(*if_function_arguments[1]->getResultType()))
+                    {
+                        DataTypePtr common_type;
+                        if (use_variant_when_no_common_type)
+                            common_type = getLeastSupertypeOrVariant(DataTypes{if_function_arguments[1]->getResultType(), function_query_node->getResultType()});
+                        else
+                            common_type = getLeastSupertype(DataTypes{if_function_arguments[1]->getResultType(), function_query_node->getResultType()});
+                        node = buildCastFunction(if_function_arguments[1], common_type, scope.context, true);
+                        resolveExpressionNode(node,
+                            scope,
+                            false /*allow_lambda_expression*/,
+                            false /*allow_table_expression*/);
+                    }
+                    else
+                        node = std::move(if_function_arguments[1]);
+
+                    return result_projection_names;
+                }
+                else
+                {
+                    bool current_do_not_execute = disable_constant_folding;
+                    disable_constant_folding = true;
+                    auto second_argument_projection_names = resolveExpressionNode(if_function_arguments[1],
+                        scope,
+                        false /*allow_lambda_expression*/,
+                        false /*allow_table_expression*/);
+                    disable_constant_folding = current_do_not_execute;
+
+                    auto second_argument_projection_name =
+                    if_function_arguments[1]->getNodeType() == QueryTreeNodeType::IDENTIFIER ?
+                        if_function_arguments[1]->as<IdentifierNode>()->getIdentifier().getFullName() :
+                        second_argument_projection_names[0];
+
+                    auto multi_if_function = std::make_shared<FunctionNode>("multiIf");
+                    for (size_t n = 2; n < if_function_arguments.size(); ++n)
+                        multi_if_function->getArguments().getNodes().push_back(std::move(if_function_arguments[n]));
+                    node = std::move(multi_if_function);
+                    auto result_projection_names = resolveFunction(node, scope);
+
+                    if (if_function_arguments[1]->getNodeType() != QueryTreeNodeType::IDENTIFIER &&
+                        node->getResultType()->equals(*if_function_arguments[1]->getResultType()))
```

Comment:
Should it be like this?
```suggestion
                        !node->getResultType()->equals(*if_function_arguments[1]->getResultType()))
```

Comparing with l 348-349

Reply:
yes